src/ChainSafe.Gaming/MultiCall/MultiCall.cs

```diff
@@ -0,0 +1,114 @@
+
+        private const string DefaultDeploymentAddress = "0xcA11bde05977b3631167028862bE2a173976CA11";
+        private readonly Contract multiCallContract;
+
+        public MultiCall(IContractBuilder builder, IChainConfig chainConfig, MultiCallConfig config)
+        {
+            if (chainConfig.ChainId != null && MultiCallDefaults.DeployedNetworks.Contains(chainConfig.ChainId))
+            {
+                multiCallContract = builder.Build(ContractAbi, DefaultDeploymentAddress);
+            }
+            else
+            {
+                if (chainConfig.ChainId != null && config.CustomNetworks.TryGetValue(chainConfig.ChainId, out var address))
+                {
+                    multiCallContract = builder.Build(ContractAbi, address);
+                }
+            }
+        }
+
+        public async Task<List<Result>> MultiCallAsync(Call3Value[] multiCalls, int pageSize = DefaultCallsPerRequest)
+        {
+             if (multiCalls.Any(x => x.Value > 0))
```

Comment:
can't this `if/else` be moved in a single foreach loop to avoid nesting and optimize code? I see a lot of common statements in both paths/conditions.

Reply:
What if the method body was something like this
```
var results = new List<object[]>();

            foreach (var page in multiCalls.Batch(pageSize))
            {
                var contractCalls = new List<Call3Value>();

                foreach (var multiCall in page)
                {
                    Call3Value call3Value = new Call3Value
                    {
                        CallData = multiCall.CallData, Target = multiCall.Target, AllowFailure = multiCall.AllowFailure,
                    };

                    if (multiCalls.Any(x => x.Value > 0))
                    {
                        call3Value.Value = multiCall.Value;
                    }

                    contractCalls.Add(call3Value);
                }

                if (contractCalls.Count <= 0)
                {
                    continue;
                }

                var callParams = new object[]
                {
                    contractCalls,
                };

                var callResults = await multiCallContract.Call(MultiCallCommonMethods.Aggregate3, callParams);

                results.Add(callResults);
            }

            return ExtractResults(results);
```

Reply:
@robGG1997 all of the items need to be one type or the other if any are found to be a Call3Value type or a regular Call3 type so there cant be a mix